### PR TITLE
fix(Curriculum): make sure the nested input element in the form is th…

### LINKED ISSUE #38478: challenge Basic HTML and HTML5: Create a Form Element can pass with wrong code
**Describe your problem and how to reproduce it:**
this challenge instruct to next the existing input element inside a form tag. It is possible to pass this challenge by creating a new input tag inside the form element without nesting the existing one.

The following code with two input elements passes:
```
<h2>CatPhotoApp</h2>
<main>
<p>Click here to view more <a href="#">cat photos</a>.</p>

<a href="#"><img src="https://bit.ly/fcc-relaxing-cat" alt="A cute orange cat lying on its back."></a>

<p>Things cats love:</p>
<ul>
  <li>cat nip</li>
  <li>laser pointers</li>
  <li>lasagna</li>
</ul>
<p>Top 3 things cats hate:</p>
<ol>
  <li>flea treatment</li>
  <li>thunder</li>
  <li>other cats</li>
</ol>
<input type="text" placeholder="cat photo URL">
<form action="/submit-cat-photo">
 <input type="text">   <!-- the issue is here, there is a new element instead of nesting ht eexisting one -->
</form>
</main>
```

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/create-a-form-element

see this topic in the forum: https://www.freecodecamp.org/forum/t/form-element-problem/366128

### PR DESCRIPTION
**Pull request by AshNaz87**
…e existing element given in the editor

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38478

### CODE REVIEW

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-form-element.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-form-element.english.md:18]:
```suggestion
Nest the existing input element inside a <code>form</code> element and add assign <code>"/submit-cat-photo"</code> to the <code>action</code> attribute of the <code>form</code> element.
```

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-form-element.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-form-element.english.md:27]:
```suggestion
  - text: The existing <code>input</code> element should be nested within a <code>form</code> element.
```

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-form-element.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-form-element.english.md:28]: Just call `document.querySelector` once:
```suggestion
    testString: const inputElem = document.querySelector('form input'); assert(inputElem.getAttribute('type') === 'text' && inputElem.getAttribute('placeholder') === 'cat photo URL');
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@AshNaz87 See the changes I have suggested.

**Comment by RandellDawson** [comment]
Once we get this approved, we will need the Chinese team to make the applicable changes to the Chinese file to match the English changes to the `text` that has been changed.  I have added a `status: blocked` label in the mean time, so we know not to merge it until the Chinese team has added a commit.

**Comment by AshNaz87** [comment]
Thanks @RandellDawson

**Review by ojeytonwilliams — DISMISSED** [review]
LGTM :+1:

**Review by hassaanp — DISMISSED** [review]

**Comment by RandellDawson** [comment]
@freeCodeCamp/chinese Since we now have 2 approvals of the English version, we need one member on the Chinese team to a commit and another member approve the commit, so we can merge this PR.

The `Instructions` section and the one `text` in the `Tests` section needs translating.

Thank you.

**Review by Manish-Giri — CHANGES_REQUESTED** [review]
Untitled

This doesn't make much sense -

> and add assign "/submit-cat-photo" to the action attribute of the form element

I'm guessing the idea was to replace "add" with "assign", perhaps. Either way, pick one.

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-form-element.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-form-element.english.md:18]:
```suggestion
Nest the existing input element inside a <code>form</code> element and assign <code>"/submit-cat-photo"</code> to the <code>action</code> attribute of the <code>form</code> element.
```

**Review by RandellDawson — DISMISSED** [review]
@AshNaz87 I have added a suggestion based on @Manish-Giri's observation.

**Review by Manish-Giri — DISMISSED** [review]

**Comment by RandellDawson** [comment]
@freeCodeCamp/chinese Still waiting on the necessary translations of this challenge.

**Comment by raisedadead** [comment]
> Still waiting on the necessary translations of this challenge.

IMHO we should create an issue and move ahead to respect everyones volunteer contributions and time?

**Comment by miyaliu666** [comment]
> @freeCodeCamp/chinese Still waiting on the necessary translations of this challenge.

I've talked with @ZhichengChen 10 hours ago. It looks like we should commit to his repo first since it hasn't been merged to master. He'll figure out how to handle it. I agree with Mrugesh we can create issues when there're more tasks like this. Thanks!

**Comment by RandellDawson** [comment]
@raisedadead I thought we agreed to keep the English and Chinese in sync.  Am I missing something.

> It looks like we should commit to his repo first since it hasn't been merged to master. He'll figure out how to handle it. 

I thought all the Responsive Web Design Chinese translation PRs have already been merged.

**Comment by raisedadead** [comment]
> I thought we agreed to keep the English and Chinese in sync. Am I missing something.

Yes we did. And after having talks with the community members (like Miya mentioned) we concluded that we would be lagging behind in the translations. This is simply because our contributors are volunteers and its a logistical issue to get in the translations right away.

We should hence track changes that need i18n as issues to avoid undue delays in the english version.

**Comment by RandellDawson** [comment]
@raisedadead I sent you message on Gitter.

**Review by amirabbaska — COMMENTED** [review]
Nice

**Comment by RandellDawson** [comment]
I resolved the merge conflict, but then the PR was failing for linting issue involving the use of a bare url being used in the markdown.  I replace `<code></code>` with single backticks and the linting passed.  I created a separate commit for the change to backticks.

**Review by RandellDawson — APPROVED** [review]
I am fine with these changes.

**Comment by RandellDawson** [comment]
@moT01 @ojeytonwilliams @nhcarrigan @Sky020 Can one of you review the PR and either approve or request changes?

Thank you.

**Review by naomi-lgbt — COMMENTED** [review]
image

This passes the tests (both before the changes in this PR and after) - is this something we want to take into consideration and account for, while we are making changes to this challenge?

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍 . Whilst I agree with @nhcarrigan that catching a case like that would be better, it is another issue, we might tackle later. The tests can also be passed by deleting the `main` element, among others.

So, we either create another test checking for changes in the seed. Or, we change the current test to ensure nothing but the `input` is within the `form`?

**Review by S1ngS1ng — APPROVED** [review]

**Comment by S1ngS1ng** [comment]
@RandellDawson Please lemme know if we need to apply the same changes to [this file](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/create-a-form-element.chinese.md). If so, I should be able to create a PR for it by EOD today.
Or instead, should we wait on merging this PR first?

**Comment by RandellDawson** [comment]
> This passes the tests (both before the changes in this PR and after) - is this something we want to take into consideration and account for, while we are making changes to this challenge?

As far as the HTML5 specification goes, you could technically leave off the closing `main` tag and it will render just fine in all modern browsers.  There are elements that **must** have closing tags (i.e h1-h6), there are elements that never have a closing tag (i.e. self-closing tags like `input` and `br`, and per the HTML5 specification most other elements (with some very specific edge cases) the closing tag is optional.   Having a matching closing tag on an element where it is optional is more about making the code more readable that a syntax requirement.  See below using @Sky020's example.  The out-of-place closing `main` tag is literally ignored (does not display in the browser) and when the DOM is constructed, a closing `main` tag is put into the proper location.

image

You could easily the closing `main` tag is optional by removing it from the HTML code:

image

We currently have many challenges where we validate the closing tags are present even when technically leaving them off would be perfectly valid HTML.  An HTML validator might possibly flag HTML with closing tags in the wrong order or missing, but we only check in those challenges to avoid trying to explain the cases where it does not matter.  Plus, as I said before, adding closing tags makes the code more readable just as indenting code does.

I said all the above to see, we do not have to worry about writing a test for an out of place closing `main` tag for this challenge.

**Comment by RandellDawson** [comment]
@S1ngS1ng Yes, let's wait until this PR is merged, then you can make the changes to the Chinese version to keep everything in sync.

Thanks.

**Comment by RandellDawson** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 🎉

**Pull request merged**
